Improvements for the Continuous Integration with OBS and GitHub/GitLab (#299)
openSUSE/obs-landing: _posts/development/2021-06-09-scm-integration-improvements.md

---
layout: post
title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
category: development
---

> Today, we have some improvements in the Continuous Integration we unveiled in our [last blog post.](https://openbuildservice.org/2021/05/31/scm-integration/).

Reviewer: There is no need to capitalize continuous integration, it's not a proper noun. [I already commented about that earlier](https://github.com/openSUSE/obs-landing/pull/299#discussion_r648976840)

```suggestion
Today, we have some improvements for the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).
```

## Updates on a PR/MR

OBS will trigger the workflow again when new changes are detected in an opened PR/MR. It will rebuild your branched packages when pushing new commits, basically.

## Build Results for Repository and Architecture

> Maybe you had a workflow set for a package built against several distributions and architectures. OBS was reporting all build results, but we were showing only the last one on the checks section of a PR/MR. Now OBS shows the build results for each repository and architecture combination of any package.

Reviewer 2: I understand what you are trying to say. The OBS shows all build results but the PR only the last one. But this is pretty hard to parse from those sentences. Maybe you can make this a bit simpler?

Author: I've reworded that paragraph. Check it out again, please

---
layout: post
title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
category: development
---

> ## Introduction

Reviewer: This header should be removed.

```suggestion
```

> Today we have some improvements over our [first attempt at Continuous Integration](https://openbuildservice.org/2021/05/31/scm-integration/).

Reviewer: ```suggestion
Today, we have some improvements for our [first iteration at continuous integration](https://openbuildservice.org/2021/05/31/scm-integration/).
```

---
layout: post
title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
category: development
---

> Today, we have some improvements for our [first iteration at continuous integration](https://openbuildservice.org/2021/05/31/scm-integration/).

Reviewer 2: The meaning of `iteration` is not what you think for your readers. This is something that makes sense for us in the team, not for anyone else.

Reviewer: I disagree with you on this one. Looking at the definition of `iteration`:

> Iteration is the repetition of a process in order to generate an outcome. The sequence will approach some end point or end value. Each repetition of the process is a single iteration, and the outcome of each iteration is then the starting point of the next iteration.

This is what we're doing. What is it for our readers if it's not that?

Author: I understand `iteration` is internal to our team's working methodology, that our users don't really care. Nevertheless I've changed that sentence so...

## Updates on a PR/MR

OBS will trigger the workflow again when new changes are detected in an opened PR/MR.

> This will allow you, for example, to rebuild branched packages when pushing new changes to the PR/MR.

Reviewer 2: You make it sound as if there is more. There isn't right?

## Build Results for Repository and Architecture

> You maybe had an integration set for a package built against several distributions and architectures. OBS was reporting all build results, but we were showing only the last one on the checks section of a PR/MR. Now OBS shows the build results for each repository and architecture combination of any package.

Reviewer 2: integration? What's that? You mean workflow?

It will appear in the PR/MR checks section as another check, with the build status reported like this:

	✓ OBS: ctris - openSUSE_Tumbleweed/x86_64
	
If your package build several flavours, the check will have the flavour name appended to the package name:

	✓ OBS: ctris:hans - openSUSE_Tumbleweed/x86_64
	
TODO: Add screenshots of an actual checks section

> ## Major SCM Vendors Support

Reviewer 2: You make this sound as if it is a feature. But it's a limitation right? :)

---
layout: post

> title: Improvements Over Continuous Integration with OBS and GitHub/GitLab

Reviewer: ```suggestion
title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
```

category: development
---
## Introduction

Last post showed the first attempt at integrating OBS and GitHub/GitLab. Today we show some improvements.

## Updates on a PR/MR

OBS will trigger the workflow again when new changes are detected in an opened PR/MR.

This will allow you to rebuild branched packages if you push new changes to a PR/MR.

> ## Build results for repository and architecture

Reviewer: The capitalization is wrong for this title: https://capitalizemytitle.com/

Author: Fixed

> Until now, we were showing only one build result. Now OBS reports the build results back to your PR/MR for each repository and architecture combination of any package.

Reviewer: This paragraph isn't true.

We reported all build results, but only the last one was displayed since we only had one status for all build results. We kept overwriting the status check. There are now one status check per build result, so this is the major difference here.

Author: Okay, what about [this](https://github.com/openSUSE/obs-landing/pull/299/files#diff-e75abc542fb8f3e86a0045bfbf08b9b9d6843c855dda1bc739e465717a98a84cR20)

Reviewer: This is fine, but the sentence above is still somehow confusing/wrong. This is what I would explain, maybe rephrase it if needed.

We reported all build results, but only the last one was displayed since we only had one status for all build results. We kept overwriting the status check.

Author: Okay, I've reworded it a bit.

It will appear in the PR/MR checks section as another check, with the build status reported like this:

	✓ OBS: ctris - openSUSE_Tumbleweed/x86_64
	

> ## Major SCM Vendors support

Reviewer: Capitalization issues here too: https://capitalizemytitle.com/

Author: Fixed

> Right now we only act on events coming from [GitHub.com](https://github.com) or [GitLab.com](https://gitlab.com).

Reviewer: ```suggestion
Right now, we only act on events coming from [GitHub.com](https://github.com) or [GitLab.com](https://gitlab.com).
```
